src/textual/css/_styles_builder.py

```diff
@@ -570,31 +571,26 @@ def process_color(self, name: str, tokens: list[Token]) -> None:
         name = name.replace("-", "_")
 
         color: Color | None = None
-        alpha: float | None = None
-
-        for token in tokens:
-            if token.name == "scalar":
-                alpha_scalar = Scalar.parse(token.value)
-                if alpha_scalar.unit != Unit.PERCENT:
-                    self.error(name, token, "alpha must be given as a percentage.")
-                alpha = alpha_scalar.value / 100.0
+        rules = self.styles._rules
 
-            elif token.name in ("color", "token"):
-                try:
-                    color = Color.parse(token.value)
-                except Exception as error:
-                    self.error(
-                        name,
-                        token,
-                        color_property_help_text(name, context="css", error=error),
-                    )
-            else:
-                self.error(name, token, color_property_help_text(name, context="css"))
+        expression = _join_tokens(tokens)
```

Comment:
We're parsing the color, combining the tokens, and re-parsing the string again.  See my suggestion re tokenize above. I think we can simplify by extracting the color from a list of tokens.